app/src/main/java/com/monstarlab/features/nstack/domain/usecase/SetupNstackUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package com.monstarlab.features.nstack.domain.usecase
+
+import com.monstarlab.core.extensions.suspendRunCatching
+import com.monstarlab.core.injection.IoDispatcher
+import dk.nodes.nstack.kotlin.NStack
+import dk.nodes.nstack.kotlin.models.AppOpenData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class SetupNstackUseCase @Inject constructor(
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) {
+
+    suspend operator fun invoke(): Result<AppOpenData> = suspendRunCatching {
+        withContext(ioDispatcher) {
+            when (val result = NStack.appOpen()) {
+                is dk.nodes.nstack.kotlin.models.Result.Success -> result.value.data
+                is dk.nodes.nstack.kotlin.models.Result.Error -> throw Exception("Failed to appOpen NStack")
```

Comment:
Can we import these guys

Reply:
No, we have `kotlin.Result` already imported and NStack use their own custom `Result` implementation.

Reply:
I'll make a custom alias